src/status_im/chat/views/geolocation/views.cljs

```diff
@@ -0,0 +1,173 @@
+                                                                       (str (or address title)
+                                                                            "&amp;" latitude
+                                                                            "&amp;" longitude)
+                                                                       false]])
+                                     (dispatch [:send-seq-argument]))}
+   [view (st/place-item-container address)
+    [view st/place-item-title-container
+     [view (st/place-item-circle-icon pin-style)]
+     [text {:style st/place-item-title
+            :number-of-lines 1
+            :font :medium}
+      title]]
+    (when address
+      [text {:style st/place-item-address
+             :number-of-lines 1}
+       address])]])
+
+(defview current-location-map-view []
+  [geolocation [:get :geolocation]
+   command     [:selected-chat-command]]
+  {:component-will-mount #(dispatch [:request-geolocation-update])
+   :component-did-mount (fn [_] (js/setTimeout #(dispatch [:chat-input-focus :seq-input-ref]) 400))}
+  (let [coord (select-keys (:coords geolocation) [:latitude :longitude])]
+    [view
+     (if (not (empty? coord))
+       [view
+        [mapview {:onTap #(do
+                            (dispatch [:set-command-argument [0 "Dropped pin" false]])
+                            (dispatch [:set-chat-seq-arg-input-text "Dropped pin"])
+                            (dispatch [:chat-input-blur :seq-input-ref])
+                            (dispatch [:load-chat-parameter-box (:command command)]))
+                  :initialCenterCoordinate coord
+                  :showsUserLocation true
+                  :initialZoomLevel 10
+                  :logoIsHidden true
+                  :rotateEnabled false
+                  :scrollEnabled false
+                  :zoomEnabled false
+                  :pitchEnabled false
+                  :style st/map-view}]]
+       [view st/map-activity-indicator-container
+        [components/activity-indicator {:animating true}]])]))
+
+(defn current-location-view []
+  (let [geolocation      (subscribe [:get :geolocation])
+        cur-loc-geocoded (r/atom nil)
+        result (reaction (when @geolocation (get-places (:coords @geolocation) cur-loc-geocoded)))]
+      (r/create-class
+        {:component-will-mount #(dispatch [:request-geolocation-update])
+         :render
+           (fn []
+             (let [_ @result]
+               (when (and @cur-loc-geocoded (> (count (:features @cur-loc-geocoded)) 0))
+                 [view st/location-container
+                  [text {:style st/location-container-title}
+                   (label :t/your-current-location)]
+                  (let [{:keys [place_name center] :as feature} (get-in @cur-loc-geocoded [:features 0])]
+                    [place-item {:title (:text feature) :address place_name :center center}])])))})))
```

Comment:
`defview` should work pretty well for this component
as well as for the ones below